scripts/release/upload_metadata_file.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+import os
+import shutil
+import subprocess
+import zipfile
+from functools import cache
+from pathlib import Path
+from typing import Any
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Upload metadata file to S3")
+    parser.add_argument(
+        "--package", type=str, required=True, help="Path to the package"
+    )
+    parser.add_argument(
+        "--bucket", type=str, required=True, help="S3 bucket to upload metadata file to"
+    )
+    parser.add_argument(
+        "--key-prefix",
+        type=str,
+        required=True,
+        help="S3 key to upload metadata file to",
+    )
+    parser.add_argument("--dry-run", action="store_true", help="Dry run")
+    args = parser.parse_args()
+    # Sanitize the input a bit by removing s3:// prefix + trailing/leading
+    # slashes
+    if args.bucket.startswith("s3://"):
+        args.bucket = args.bucket[5:]
+    if args.bucket.endswith("/"):
+        args.bucket = args.bucket[:-1]
+    if args.key_prefix.startswith("/"):
+        args.key_prefix = args.key_prefix[1:]
+    if args.key_prefix.endswith("/"):
+        args.key_prefix = args.key_prefix[:-1]
+    return args
+
+
+@cache
+def get_s3_client() -> Any:
```

Comment:
Not entirely sure I like this method of installing dependencies, is there a better way of doing this?